pallets/asset/src/lib.rs

```diff
@@ -704,30 +690,14 @@ decl_module! {
         /// - `InsufficientPortfolioBalance` If the PIA's default portfolio doesn't have enough free balance
         #[weight = <T as Trait>::WeightInfo::redeem()]
         pub fn redeem(origin, ticker: Ticker, value: T::Balance) {
-            let PermissionedCallOriginData {
-                primary_did: did,
-                secondary_key,
-                ..
-            } = Identity::<T>::ensure_origin_call_permissions(origin)?;
-
-            // Ensure that the secondary key has asset permission
-            Self::ensure_asset_perms(secondary_key.as_ref(), &ticker)?;
+            // Ensure origin is PIA with custody and permissions for default portfolio.
+            let pia = Self::ensure_pia_or_owner_with_custody(origin, ticker)?.primary_did;
```

Comment:
This assumption never stands true always. or we assume that only pia is only allowed to call the redeem function.

Reply:
See above.